addons/website_event/models/website_visitor.py

```diff
@@ -21,6 +21,16 @@ class WebsiteVisitor(models.Model):
         search="_search_event_registered_ids",
         groups="event.group_event_registration_desk")
 
+    @api.depends('partner_id, event_registration_ids.name')
+    def name_get(self):
+        """ If there is an event registration for an anonymous visitor, use that
+        registered attendee name as visitor name. """
+        res_dict = dict(super().name_get())
+        # sudo is needed for `event_registration_ids`
+        for visitor in self.sudo().filtered(lambda v: not v.partner_id and v.event_registration_ids):
+            res_dict[visitor.id] = visitor.event_registration_ids[-1].name
+        return list(res_dict.items())
```

Comment:
Are you sure ordering is kept ? Doing a list -> dict -> list ensure ordering is always the same ?